#660 implemented AbbreviationsAsWordInNameCheck

original-brownbear opened the pull request:
#660 AbbreviationAsWordInNameCheck configured:
* Set allowed abbreviation size to 1 so to not allow those
* Set "IT" as the only exception as allowed uppercase abbreviation
* Static final fields as well as overrides are not checked by default and needed no config

Added tests for:
* Abbreviations XML/HTML are detected as standard violations
* IT is not a violation
* Static final fields are not checked
* Correctly named classes pass the check
* Overrides are not checked

krzyk commented:
@original-brownbear why didn't you use the check from checkstyle?

krzyk commented:
@original-brownbear Please use AbbreviationAsWordInNameCheck from checkstyle, and configure it to exclude IT abbreviation, final static fields, overriden methods.

davvd commented:
@original-brownbear Thanks, someone will review your pull request soon

davvd commented:
@dmzaytsev review this one, please

original-brownbear commented:
@krzyk on it, sorry misunderstood the setup a little all clear now :)

original-brownbear commented:
@dmzaytsev implemented this using the checkstyle functionality now.

Added tests for the requested cases.
Set allowed abbreviation size to 1 so to not allow those.
Set "IT" as the only exception.
The final static fields case is covered by the default settings as far as I can see, added test for this too.

krzyk commented:
@original-brownbear BTW. If you have spare time, consider http://at.teamed.io/join.html

dmzaytsev commented:
@original-brownbear thank you
also please update the pull request description 
you have reverted the renames back, right?
>Renamed suffixes ITCase and IT to ItCase and It to bring them in line with the new check

dmzaytsev commented:
@rultor merge

rultor commented:
> @rultor merge

@dmzaytsev Thanks for your request. @krzyk Please confirm this.

krzyk commented on qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java:
```diff
@@ -584,6 +584,65 @@ public void rejectsSpacesInsideMethods() throws Exception {
     }
 
     /**
+     * CheckstyleValidator can reject uppercase Abbreviations in naming
```
@original-brownbear lower case for `Abbreviations`

krzyk commented on qulice-checkstyle/src/test/resources/com/qulice/checkstyle/ValidAbbreviationAsWordInNameIT.java:
```diff
@@ -0,0 +1,18 @@
+ */
+public class ValidAbbreviationAsWordInNameIT {
+    /**
+     * Valid name on inner class, because the IT abbreviation is allow.
```
@original-brownbear *... is allowed.*

krzyk commented:
@original-brownbear two minor comments from me, and as @dmzaytsev wrote, please update the description

original-brownbear commented:
@krzyk pushed the requested changes, updated pull request description (force pushed onto it to remove the initial solution way missing the point, so the "IT" -> "It" changes are not in this).

krzyk commented:
@original-brownbear you missed one of my comments

original-brownbear commented:
@krzyk sorry my bad, I mistakenly jumped a line and thought it referred to the somewhat verbose Javadoc above. Fixed the typo now :)

krzyk commented:
@davvd this is not postponed

krzyk commented:
@rultor merge pls

rultor commented:
> @rultor merge pls

@krzyk OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/6980-183632156)

rultor commented:
> @rultor merge pls

@krzyk Done! FYI, the full log is [here](http://www.rultor.com/t/6980-183632156) (took me 20min)

davvd commented:
> @davvd this is not postponed

@krzyk right, I removed "postponed" tag

davvd commented:
@dmzaytsev quality is good here, 10 mins added to @krzyk (our architect) in transaction 77677180. **17 mins** sent to your balance (ID `77677204`), many thanks! It took 74 hours and 8 mins.. review comments (c=2) added as a bonus. +17 added to your rating, at the moment it is: [+3233](http://www.netbout.com/b/35166?open=rating)

davvd commented:
@rultor deploy

rultor commented:
> @rultor deploy

@davvd OK, I'll try to deploy now. You can check the progress [here](http://www.rultor.com/t/6980-184674980)

rultor commented:
> @rultor deploy

@davvd Done! FYI, the full log is [here](http://www.rultor.com/t/6980-184674980) (took me 36min)
